tests/e2e_tests/l2b_to_l3_e2e.py

```diff
@@ -0,0 +1,357 @@
+            
+            # Verify HDU count
+            with fits.open(l3_filename) as hdul:
+                verify_hdu_count(hdul, 5, frame_info, logger)  # L3 should have 5 HDUs
+            
+            # Verify data level
+            verify_header_keywords(img.ext_hdr, {'DATALVL': 'L3'}, frame_info, logger)
+            
+            # Check data dimensions
+            if is_polarimetry:
+                # Polarimetry data should be a datacube (2, image_size, image_size)
+                if len(img.data.shape) == 3 and img.data.shape[0] == 2:
+                    logger.info(f"{frame_info}: Polarimetry datacube shape {img.data.shape}. PASS")
+                else:
+                    logger.info(f"{frame_info}: Expected polarimetry datacube (2, N, N), got {img.data.shape}. FAIL")
+            else:
+                # Standard imaging data should be 2D
+                check_dimensions(img.data, (125,125), frame_info, logger)
+            
+            # Verify WCS headers exist (from create_wcs step)
+            wcs_keys = ['CRVAL1', 'CRVAL2', 'CRPIX1', 'CRPIX2', 'CTYPE1', 'CTYPE2']
+            missing_wcs = [k for k in wcs_keys if k not in img.ext_hdr]
+            if not missing_wcs:
+                logger.info(f"{frame_info}: WCS headers present ({', '.join(wcs_keys)}). PASS")
+            else:
+                logger.info(f"{frame_info}: WCS headers incomplete. Missing: {', '.join(missing_wcs)}). FAIL")
+            
+            # Verify data has been divided by exposure time (should be in photoelectrons/s)
+            if img.ext_hdr['BUNIT'] == 'photoelectron/s':
+                logger.info(f"{frame_info}: BUNIT = 'photoelectron/s'. PASS")
+            else:
+                logger.info(f"{frame_info}: BUNIT = '{img.ext_hdr['BUNIT']}'. Expected: 'photoelectron/s'. FAIL")
+            
+        except Exception as e:
+            logger.info(f"{frame_info}: Validation failed with error: {str(e)}. FAIL")
+            raise
+        
+        logger.info('')
+    
+    logger.info(f"Total output L3 images validated: {len(new_l3_filenames)}")
+    logger.info('')
+    
+    # remove temporary caldb file
+    os.remove(tmp_caldb_csv)
+    
+    return new_l3_filenames
+
+
+@pytest.mark.e2e
+def test_l2b_to_l3(e2edata_path, e2eoutput_path, input_datadir, cals_dir):
+    """Run the complete L2b to L3 end-to-end test.
+    
+    Args:
+        e2edata_path (str): Path to test data
+        e2eoutput_path (str): Output directory path for results and logs
+        input_datadir (str or None): Custom input data directory
+        cals_dir (str or None): Custom calibration directory
+    """
+    # Set up output directory and logging
+    global logger
+    
+    # Use custom paths if provided, otherwise fall back to defaults from e2edata_path
+    if input_datadir is None:
+        l2b_datadir = os.path.join(e2edata_path, "SPEC_targetstar_slit_prism", "L2b")
```

Comment:
how are these L2b data produced? Did you produce them using the l1_to_l2b e2e test? If so, I think we should chain this test to use the on-the-fly output of the l1_to_l2b_e2e test to be used as the input here.